scripts/resetdb.sh

```diff
@@ -1,16 +1,59 @@
 #!/bin/bash
 
-readonly TRILLIAN_PATH=$(go list -f '{{.Dir}}' github.com/google/trillian)
-
-echo "Completely wipe and reset database 'test'."
-read -p "Are you sure? " -n 1 -r
-if [[ $REPLY =~ ^[Yy]$ ]]
-then
-    # User-supplied arguments must be first. This is because some flags, such
-    # as --defaults-extra-file, must be at the start.
-    mysql "$@" -u root -e 'DROP DATABASE IF EXISTS test;'
-    mysql "$@" -u root -e 'CREATE DATABASE test;'
-    mysql "$@" -u root -e "GRANT ALL ON test.* TO 'test'@'localhost' IDENTIFIED BY 'zaphod';"
-    mysql "$@" -u root -D test < ${TRILLIAN_PATH}/storage/mysql/storage.sql
-fi
-echo
+set -e
+
+usage() {
+  echo "$0 [--force] [--verbose] ..."
+  echo "accepts environment variables:"
+  echo " - DB_NAME"
+  echo " - DB_USER"
+  echo " - DB_PASSWORD"
+}
+
+collect_vars() {
+  # set unset environment variables to defaults
+  [ -z ${DB_USER+x} ] && DB_USER="root"
+  [ -z ${DB_NAME+x} ] && DB_NAME="test"
+  # format reused supplied environment variables
+  FLAGS=""
+  [ -z ${DB_PASSWORD+x} ] || FLAGS="${FLAGS} -p$DB_PASSWORD"
+
+  # handle flags
+  FORCE=false
+  VERBOSE=false
+  while [[ $# -gt 0 ]]; do
+    case "$1" in
+      --force) FORCE=true ;;
+      --verbose) VERBOSE=true ;;
+      *) FLAGS="${FLAGS} $1"
```

Comment:
Suggested fix: https://github.com/RJPercival/trillian/commits/witoff/aws

This will properly handle quoted arguments. I made another couple of small improvements while I was poking around in that file, which you can find just after that commit.